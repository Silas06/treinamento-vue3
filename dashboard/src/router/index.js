import { createRouter, createWebHistory } from 'vue-router'

const Feedbacks = () => import('@/views/Feedbacks')
const Credencials = () => import('@/views/Credencials')
const Home = () => import('@/views/Home')

export const routes = [
  {
    path: '/',
    name: 'Home',
    component: Home
  },
  {
    path: '/credencials',
    name: 'Credencials',
    component: Credencials,
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/feedbacks',
    name: 'Feedbacks',
    component: Feedbacks,
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/:pathMatch(.*)*',
    redirect: { name: 'Home' }
  }
]

const router = createRouter({
  history: createWebHistory(''),
  routes
})

export default router
