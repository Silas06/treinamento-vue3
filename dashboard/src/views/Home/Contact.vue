<template>
  <div class=" w-full justify-center flex">
    <div class="flex flex-col items-center w-4/5 max-2-6lg my-16">
      <h1 class="text-4xl font-black text-center text-gray-800">
        Alguma dúvida?
      </h1>
      <p claas="text-lg text-gray-800 font-regular">
        Quer saber melhor como funciona e quais são os preços?
      </p>
      <div class="mt-10">
      <a href="mailto:" class="px-6 mt-10 py-2 font-bold text-white rounded-full bg-brand-main focus:outline-none">
      Nos mande um e-mail!
      </a>
      </div>
    </div>
  </div>
</template>

<script>
export default {}
</script>

<style></style>
