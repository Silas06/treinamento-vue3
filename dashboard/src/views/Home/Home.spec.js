import Home from '.'
import { routes } from '../../router'
import { shallowMount } from '@vue/test-utils'

import { createRouter, createWebHistory } from 'vue-router'

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

describe('<Home />', () => {
  it('Abrir pagina Home correta', async () => {
    router.push('/')
    await router.isReady()
    const wrapper = shallowMount(Home, {
      global: {
        plugins: [router]
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
