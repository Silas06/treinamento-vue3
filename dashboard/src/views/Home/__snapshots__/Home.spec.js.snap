// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Abrir pagina Home correta 1`] = `
<custom-header-stub></custom-header-stub>
<contact-stub></contact-stub>
<div class="flex justify-center py-10 bg-brand-gray">
  <p class="font-medium text-gray-800 text-center">feedbacker Â© 2021</p>
</div>
`;
